test/testchain/chain.go

```diff
@@ -0,0 +1,216 @@
+	genesis *genesis.Genesis,
+	engine bft.Committer,
+	repo *chain.Repository,
+	stater *state.Stater,
+	genesisBlock *block.Block,
+	logDB *logdb.LogDB,
+) *Chain {
+	return &Chain{
+		db:           db,
+		genesis:      genesis,
+		engine:       engine,
+		repo:         repo,
+		stater:       stater,
+		genesisBlock: genesisBlock,
+		logDB:        logDB,
+	}
+}
+
+// NewIntegrationTestChain is a convenience function that creates a Chain for testing.
+// It uses an in-memory database, development network genesis, and a solo BFT engine.
+func NewIntegrationTestChain() (*Chain, error) {
+	// Initialize the database
+	db := muxdb.NewMem()
+
+	// Create the state manager (Stater) with the initialized database.
+	stater := state.NewStater(db)
+
+	// Initialize the genesis and retrieve the genesis block
+	gene := genesis.NewDevnet()
+	geneBlk, _, _, err := gene.Build(stater)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the repository which manages chain data, using the database and genesis block.
+	repo, err := chain.NewRepository(db, geneBlk)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create an inMemory logdb
+	logDb, err := logdb.NewMem()
+	if err != nil {
+		return nil, err
+	}
+
+	return New(
+		db,
+		gene,
+		solo.NewBFTEngine(repo),
+		repo,
+		stater,
+		geneBlk,
+		logDb,
+	), nil
+}
+
+// Repo returns the blockchain's repository, which stores blocks and other data.
+func (c *Chain) Repo() *chain.Repository {
+	return c.repo
+}
+
+// Stater returns the current state manager of the chain, which is responsible for managing the state of accounts and other elements.
+func (c *Chain) Stater() *state.Stater {
+	return c.stater
+}
+
+// Engine returns the consensus engine responsible for the blockchain's consensus mechanism.
+func (c *Chain) Engine() bft.Committer {
+	return c.engine
+}
+
+// GenesisBlock returns the genesis block of the chain, which is the first block in the blockchain.
+func (c *Chain) GenesisBlock() *block.Block {
+	return c.genesisBlock
+}
+
+// MintTransactions creates a block with the provided transactions and adds it to the blockchain.
+// It wraps the transactions with receipts and passes them to MintTransactionsWithReceiptFunc.
+func (c *Chain) MintTransactions(account genesis.DevAccount, transactions ...*tx.Transaction) error {
+	return c.MintBlock(account, transactions...)
+}
+
+// MintBlock creates and finalizes a new block with the given transactions.
+// It schedules a new block, adopts transactions, packs them into a block, and commits it to the chain.
+func (c *Chain) MintBlock(account genesis.DevAccount, transactions ...*tx.Transaction) error {
+	// Create a new block packer with the current chain state and account information.
+	blkPacker := packer.New(c.Repo(), c.Stater(), account.Address, &genesis.DevAccounts()[0].Address, thor.NoFork)
+
+	// Schedule a new block to be packed at the current time.
+	blkFlow, err := blkPacker.Schedule(c.Repo().BestBlockSummary(), uint64(time.Now().Unix()))
```

Comment:
Would you consider `*packer.Mock(parent *chain.BlockSummary, targetTime uint64, gasLimit uint64)` would be better here, it will skip proposer check.